todos/src/App.tsx

```diff
@@ -0,0 +1,67 @@
+import { useState, useRef } from 'react';
+import './App.css';
+import {OptionBar} from './Components/OptionBar';
+import TodosList from './Components/TodosList';
+import {todo} from './Types/todo'
+
+export function App() {
+
+  function useTodos(){
+
+    const submitCompleteTodo = (currID: string) => {
+      setAllTodos(allTodos.map(todo => todo.id === currID ? {id: currID, label: todo.label, active: !todo.active} : todo ));
+     }
+
+     const submitInProgressTodo = (currID: string) => {
+      setAllTodos(allTodos.map(todo =>  todo.id === currID ? {id: currID, label: todo.label, active: !todo.active} : todo));
+     }
+   
+     const addTodo = () => {
+       const todoToAdd = {id: allTodos.length.toString(), label:enteredTodoLabel.current!.value, active: true};
+       setAllTodos ( allTodos.concat(todoToAdd))
```

Comment:
Are you sure this works state setters need new value and this line mutates an existing array. Pls check